.github/workflows/pre-merge-python.yml

```diff
@@ -46,10 +46,19 @@ jobs:
         env:
           GITHUB_TOKEN: ${{ secrets.CI_TOKEN }}
         run: |
-          ./container/build.sh --tag ${{ steps.define_image_tag.outputs.image_tag }} --target ci_minimum --framework ${{ matrix.framework }}
-      - name: Run Rust checks (llm/block-manager)
+          ./container/build.sh --tag ${{ steps.define_image_tag.outputs.image_tag }} --target dev --framework ${{ matrix.framework }}
+      - name: Start services with docker-compose
+        working-directory: ./deploy
         run: |
-          docker run -w /workspace/lib/llm --name ${{ env.CONTAINER_ID }}_rust_checks ${{ steps.define_image_tag.outputs.image_tag }} bash -ec 'rustup component add rustfmt clippy && cargo fmt -- --check && cargo clippy --features block-manager --no-deps --all-targets -- -D warnings && cargo test --locked --all-targets --features=block-manager'
+          docker compose up -d nats-server etcd-server
+      - name: Run Rust checks (block-manager + integration tests)
+        run: |
+          docker run -w /workspace/lib/llm --name ${{ env.CONTAINER_ID }}_rust_checks ${{ steps.define_image_tag.outputs.image_tag }} bash -ec 'rustup component add rustfmt clippy && cargo fmt -- --check && cargo clippy --features block-manager --no-deps --all-targets -- -D warnings && cargo test --locked --all-targets --features=block-manager && cargo test --locked --features integration -- --nocapture'
```

Comment:
What's the general plan here on testing rust things inside dynamo container (pre-merge-python [should be renamed]) and vs outside container (pre-merge-rust)?

My understanding was that only the KVBM tests were put in here because `pip install nixl` didn't have the necessary NIXL shared libraries to test it outside of the container.

For everything else on the rust test side, it's pretty lightweight and runs in a vanilla ubuntu container with nothing but cargo: https://github.com/ai-dynamo/dynamo/blob/faafa5ffe67c154597e8c68fbc565d68aedb2da2/.github/workflows/pre-merge-rust.yml#L39